nevergrad/functions/images/core.py

```diff
@@ -97,37 +129,79 @@ def __init__(self, classifier: nn.Module, image: torch.Tensor, label: int = 0, t
         # classifier and image cant be set as descriptors
         self.add_descriptors(label=label, targeted=targeted, epsilon=epsilon)
 
+    def _loss(self, x: np.ndarray) -> float:
+        output_adv = self._get_classifier_output(x)
+        value = float(self.criterion(output_adv, self.label).item())
+        return value * (1.0 if self.targeted else -1.0)
+
+    def _get_classifier_output(self, x: np.ndarray) -> tp.Any:
+        # call to the classifier given the input array
+        y = torch.Tensor(x)
+        image_adv = torch.clamp(self.image + y, 0, 1)
+        image_adv = image_adv.view(1, 3, self.imsize, self.imsize)
+        return self.classifier(image_adv)
+
+    # pylint: disable=arguments-differ
+    def evaluation_function(self, x: np.ndarray) -> float:  # type: ignore
+        """Returns wether the attack worked or not
+        """
+        output_adv = self._get_classifier_output(x)
+        _, pred = torch.max(output_adv, axis=1)
+        actual = int(self.label)
+        return float(pred == actual if self.targeted else pred != actual)
```

Comment:
@LaurentMnr95 this should do what you want for the final evaluation -> either 1 or 0 depending if the attack is successful or not